packages/backend/src/functions/oauth/GitHub.ts

```diff
@@ -136,6 +137,19 @@ export default class GitHub {
       return res.sendStatus(401);
     }
 
+    try {
+      // Register this session in db
+      await addSession(userId, user.id);
+    } catch (e) {
```

Comment:
Can this be avoided? One should avoid programming normal program logic with exceptions.

Reply:
I wouldn't say it's logic with exceptions, it's an exception from the database and the mongo driver exposes this as a javascript error. There's really no other way other than querying the database twice which would be an anti-pattern; we can rely on mongo ensuring the uniqueness property and should use this functionality.

We could move the try/catch into the addSession functionality and simply re-throw the exception if it is another exception than 11000. Maybe that'd be better to avoid leaking internal details of the session?

Reply:
Sounds reasonable